src/EFCore.PG/Migrations/NpgsqlMigrationsSqlGenerator.cs

```diff
@@ -757,6 +769,81 @@ protected virtual void GenerateDropEnum(PostgresEnum @enum, MigrationCommandList
                 .AppendLine(";");
         }
 
+        #endregion Enum management
+
+        #region Range management
+
+        protected virtual void GenerateRangeStatements(
+                AlterDatabaseOperation operation,
+                IModel model,
+                MigrationCommandListBuilder builder)
+        {
+            foreach (var rangeTypeToCreate in PostgresRange.GetPostgresRanges(operation)
+                .Where(ne => PostgresRange.GetPostgresRanges(operation.OldDatabase).All(oe => oe.Name != ne.Name)))
+            {
+                GenerateCreateRange(rangeTypeToCreate, model, builder);
+            }
+
+            foreach (var rangeTypeToDrop in PostgresRange.GetPostgresRanges(operation.OldDatabase)
+                .Where(oe => PostgresRange.GetPostgresRanges(operation).All(ne => ne.Name != oe.Name)))
+            {
+                GenerateDropRange(rangeTypeToDrop, builder);
+            }
+
+            foreach (var rangeTypeToAlter in from newRange in PostgresRange.GetPostgresRanges(operation)
+                join oldRange in PostgresRange.GetPostgresRanges(operation.OldDatabase) on newRange.Name equals oldRange.Name
```

Comment:
This is a complicated looking loop header. Why not go full LINQ and use `.ToList().ForEach(...)`?

Reply:
Converted to a different syntax more fitting this check.